src/__tests__/with-theme.js

```diff
@@ -0,0 +1,71 @@
+/* eslint func-style:0 */
+import React, {Component} from 'react'
+import {render, mount} from 'enzyme'
+import * as jestGlamorReact from 'jest-glamor-react'
+
+import withTheme from '../with-theme'
+import ThemeProvider, {CHANNEL} from '../theme-provider'
+
+expect.extend(jestGlamorReact.matcher)
+expect.addSnapshotSerializer(jestGlamorReact.serializer)
+
+const getMockedContext = unsubscribe => ({
+  [CHANNEL]: {
+    getState: () => {},
+    setState: () => {},
+    subscribe: () => unsubscribe,
+  },
+})
+
+test('renders a non-glamorous component with theme', () => {
+  const CompWithTheme = withTheme(({theme: {padding}}) => (
+    <div style={{padding}} />
+  ))
+  expect(
+    render(
+      <ThemeProvider theme={{padding: '10px'}}>
+        <CompWithTheme />
+      </ThemeProvider>,
+    ),
+  ).toMatchSnapshotWithGlamor()
+})
+
+test('theme properties updates get propagated down the tree', () => {
+  class Parent extends Component {
+    state = {
+      padding: 10,
+    }
+
+    render() {
+      return (
+        <ThemeProvider theme={{padding: this.state.padding}}>
+          <Child />
+        </ThemeProvider>
+      )
+    }
+  }
+
+  const Child = withTheme(({theme: {padding}}) => <div style={{padding}} />)
+  const wrapper = mount(<Parent />)
+  expect(wrapper).toMatchSnapshotWithGlamor(`with theme prop of padding 10px`)
+  wrapper.setState({padding: 20})
+  expect(wrapper).toMatchSnapshotWithGlamor(`with theme prop of padding 20px`)
+})
+
+test('acts as pass through when no theme provider found up tree', () => {
```

Comment:
i had a console.warn statement in at first but it broke linting (not that it can't be ignored). I don't think you can really tell if you have a provider, but you can definitely warn if there is no `__glamorous__` context prop (which is a good indicator of that situation).